ServerCore/Pages/Puzzles/Checklist.cshtml.cs

```diff
@@ -73,7 +79,28 @@ public async Task<IActionResult> OnGetAsync()
                 puzzleAnswers.TryGetValue(puzzle.ID, out ContentFile puzzleAnswer);
                 puzzlePrereqs.TryGetValue(puzzle.ID, out List<string> prereqs);
                 puzzleResponses.TryGetValue(puzzle.ID, out ResponseData responses);
-                puzzleHints.TryGetValue(puzzle.ID, out int hints);
+                puzzleHints.TryGetValue(puzzle.ID, out IEnumerable<int> hints);
+
+                int totalHintCostThisPuzzle = 0;
+                int hintsCountThisPuzzle = 0;
+                if (hints != null)
+                {
+                    int minHintCostThisPuzzle = 0;
+
+                    hintsCountThisPuzzle = hints.Count();
+                    foreach (int cost in hints)
+                    {
+                        minHintCostThisPuzzle = Math.Min(minHintCostThisPuzzle, cost);
+                    }
+
+                    totalHintCostThisPuzzle = Math.Abs(minHintCostThisPuzzle);
```

Comment:
the minHintCost will always be non-positive so this can be just `-` instead of `Math.Abs`. I think that's more clear.